pilot/pkg/xds/filters/filters.go

```diff
@@ -306,6 +284,31 @@ var (
 	}
 )
 
+func BuildSidecarOutboundMetadataFilter(stripHeaders bool) *hcm.HttpFilter {
```

Comment:
This is used a lot - it should be pre-computed, not generated every time.

Reply:
I could see maybe pre-computing for every proxy - but I'm not sure if it can be done pre-computed globally for the entire istiod.  The current design implementation is to use proxyHeaders API, and based on John's first pass, this seems like the best place to plug in the flag from proxyconfig:
https://github.com/istio/istio/pull/46429/files#diff-52cfa21853af325e375479ea6e0046ca6c982f094727bd3b623d066ecf966c51R701

Reply:
just created two `SidecarOutboundMetadataFilter` objects, one disabled and one enabled